recipe/meta.yaml

```diff
@@ -22,7 +22,8 @@ requirements:
   host:
     - pip
     - python
-    - hatchling >=0.25
+    - hatchling
```

Comment:
I still see hatchling pinning required.
https://github.com/jupyter/jupyter_client/blob/v7.4.5/pyproject.toml#L2

But I am fine with pushing it like this, given that our oldest hatchling is 1.8.0